src/Common/CurrentMemoryTracker.cpp

```diff
@@ -57,6 +57,7 @@ AllocationTrace CurrentMemoryTracker::allocImpl(Int64 size, bool throw_if_memory
             {
                 auto res = memory_tracker->allocImpl(will_be, throw_if_memory_exceeded);
                 current_thread->untracked_memory = 0;
+                current_thread->updateUntrackedMemoryLimit(memory_tracker->get());
```

Comment:
(Suggestion). I understand the strategy of memory enhancing of this PR. As I understand, if we have several hundreds allocations of several KB we are going to flush it a lot of times. If this part is a perfomance-sensitive one, you can replace the current flushing strategy with the exponential one (with exponent 2, for example) goind from min_untracked_memory to max_untracked memory. This will allow us to have log_2(max_untracked_memory/min_untracked_memory) flushes to the main memory tracker until max_untracked_memory is not riched

Reply:
It is exponential, but not with factor 2. The factor is 1.0625 (where 6.25% = 1/16). The idea is to keep total untracked memory to be less than 10% of total memory.